cuda/test/solver/upper_trs_kernels.cpp

```diff
@@ -0,0 +1,180 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/solver/upper_trs.hpp>
+
+
+#include <memory>
+#include <random>
+
+
+#include <cuda.h>
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+
+#include "core/solver/upper_trs_kernels.hpp"
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+class UpperTrs : public ::testing::Test {
+protected:
+    using CsrMtx = gko::matrix::Csr<double, gko::int32>;
+    using Mtx = gko::matrix::Dense<>;
+
+    UpperTrs() : rand_engine(30) {}
+
+    void SetUp()
+    {
+        ASSERT_GT(gko::CudaExecutor::get_num_devices(), 0);
+        ref = gko::ReferenceExecutor::create();
+        cuda = gko::CudaExecutor::create(0, ref);
+    }
+
+    void TearDown()
+    {
+        if (cuda != nullptr) {
+            ASSERT_NO_THROW(cuda->synchronize());
+        }
+    }
+
+    std::unique_ptr<Mtx> gen_mtx(int num_rows, int num_cols)
+    {
+        return gko::test::generate_random_matrix<Mtx>(
+            num_rows, num_cols,
+            std::uniform_int_distribution<>(num_cols, num_cols),
+            std::normal_distribution<>(-1.0, 1.0), rand_engine, ref);
+    }
+
+    std::unique_ptr<Mtx> gen_u_mtx(int num_rows, int num_cols)
+    {
+        return gko::test::generate_random_upper_triangular_matrix<Mtx>(
+            num_rows, num_cols, false,
+            std::uniform_int_distribution<>(num_cols, num_cols),
+            std::normal_distribution<>(-1.0, 1.0), rand_engine, ref);
+    }
+
+    void initialize_data(int m, int n)
+    {
+        mtx = gen_u_mtx(m, m);
+        b = gen_mtx(m, n);
+        x = gen_mtx(m, n);
+        csr_mtx = CsrMtx::create(ref);
+        mtx->convert_to(csr_mtx.get());
+        d_csr_mtx = CsrMtx::create(cuda);
+        d_x = Mtx::create(cuda);
+        d_x->copy_from(x.get());
+        d_csr_mtx->copy_from(csr_mtx.get());
+        b2 = Mtx::create(ref);
+        d_b2 = Mtx::create(cuda);
+        d_b2->copy_from(b.get());
+        b2->copy_from(b.get());
+    }
+
+    std::shared_ptr<Mtx> b;
+    std::shared_ptr<Mtx> b2;
+    std::shared_ptr<Mtx> x;
+    std::shared_ptr<Mtx> mtx;
+    std::shared_ptr<CsrMtx> csr_mtx;
+    std::shared_ptr<Mtx> d_b;
+    std::shared_ptr<Mtx> d_b2;
+    std::shared_ptr<Mtx> d_x;
+    std::shared_ptr<CsrMtx> d_csr_mtx;
+    std::shared_ptr<gko::ReferenceExecutor> ref;
+    std::shared_ptr<const gko::CudaExecutor> cuda;
+    std::ranlux48 rand_engine;
+};
+
+
+TEST_F(UpperTrs, CudaUpperTrsFlagCheckIsCorrect)
+{
+    bool trans_flag = true;
+    bool expected_flag = false;
+
+
+#if (defined(CUDA_VERSION) && (CUDA_VERSION < 9020))
+
+
+    expected_flag = true;
+
+
+#endif
```

Comment:
nit: Here, I would actually suggest to remove the newlines since only a single line is enclosed (would be easier to read that way in my opinion).